src/scanner.c

```diff
@@ -295,10 +295,11 @@ ts_scanner_next(ScannerCtx *ctx, InternalScannerCtx *ictx)
 #if PG12_GE
 				TupleTableSlot *slot = ictx->tinfo.slot;
 
-				PushActiveSnapshot(GetLatestSnapshot());
+				Assert(ctx->snapshot);
+				PushActiveSnapshot(ctx->snapshot);
```

Comment:
I am actually not sure we need to set the active snapshot here. We're passing the snapshot to `table_tuple_lock` so it's already instructed to use that snapshot and I don't see any other operations happening before we pop it that would need an active snapshot.

Reply:
Good point. Investigating.

Reply:
I changed the code. The only case we do not cover with this is somebody want to create a scanner with one snapshot and push another one temporarily for the actual scan. This does not seem sensible and you can temporarily set the snapshot yourself in that case, so I do not think this is something we should bother about.